Tweetinvi.Controllers/Tweet/TweetQueryGenerator.cs

```diff
@@ -120,21 +120,38 @@ public string GetPublishRetweetQuery(long tweetId)
         }
 
         // Get Retweets
-        public string GetRetweetsQuery(ITweetDTO tweetDTO)
+        public string GetRetweetsQuery(ITweetDTO tweetDTO, int maxRetweetsToRetrieve)
         {
             if (!_tweetQueryValidator.IsTweetPublished(tweetDTO))
             {
                 return null;
             }
 
-            return GetRetweetsQuery(tweetDTO.Id);
+            return GetRetweetsQuery(tweetDTO.Id, maxRetweetsToRetrieve);
         }
 
-        public string GetRetweetsQuery(long tweetId)
+        public string GetRetweetsQuery(long tweetId, int maxRetweetsToRetrieve)
         {
-            return string.Format(Resources.Tweet_Retweet_GetRetweets, tweetId);
+            return string.Format(Resources.Tweet_Retweet_GetRetweets, tweetId, maxRetweetsToRetrieve);
         }
-        
+
+        #region Get Retweeter Ids
+        public string GetRetweeterIdsQuery(ITweetIdentifier tweetIdentifier, bool stringifyIds)
+        {
+            if (!_tweetQueryValidator.IsValidTweetIdentifier(tweetIdentifier))
+            {
+                return null;
+            }
+            return GetRetweeterIdsQuery(tweetIdentifier.Id, stringifyIds);
+        }
+
+        public string GetRetweeterIdsQuery(long tweetId, bool stringifyIds)
+        {
+            return string.Format(Resources.Tweet_GetRetweeters, tweetId, stringifyIds.ToString());

```

Comment:
The way to buidl query has also changed, you should use `StringBuilder.AddParameterToQuery` or `StringBuilder.Append` instead of having the Resources include `{0}` values. This allow a better readability.